File: 2020/csharp/Aoc2020.Tests/Day13Tests.cs
using System.Linq;
using System.Collections.Generic;
using Xunit;

namespace Aoc2020.Tests
{
    public class Day13Tests
    {
        [Fact]
        public void DepartureInfoWorks()
        {
            var result = Day13.DepartureInfo(Utils.ReadInputLines("13-test"));
            Assert.Equal((59, 5, 295), result);
        }

        // [Theory]
        // [InlineData(1068781L, "7,13,x,x,59,x,31,19")]
        // [InlineData(3417L, "17,x,13,19")]
        // [InlineData(754018L, "67,7,59,61")]
        // [InlineData(779210L, "67,x,7,59,61")]
        // [InlineData(1261476L, "67,7,x,59,61")]
        // [InlineData(1202161486L, "1789,37,47,1889")]
        // public void GoldWorks(long result, string input)
        // {
        //     Assert.Equal(result, Day13.Gold(input));
        // }

        [Fact]
        public void Stars()
        {
            var run = new Day13();
            Assert.Equal("3966", run.Result1());
            // Assert.Equal("", run.Result2());
        }
    }
}

File: 2022/fsharp/Aoc2022.Tests/Day13.Tests.fs
module Day13.Tests

open Xunit
open Utils
open Day13

let input =
    """
    [1,1,3,1,1]
    [1,1,5,1,1]

    [[1],[2,3,4]]
    [[1],4]

    [9]
    [[8,7,6]]

    [[4,4],4,4]
    [[4,4],4,4,4]

    [7,7,7,7]
    [7,7,7]

    []
    [3]

    [[[]]]
    [[]]

    [1,[2,[3,[4,[5,6,7]]]],8,9]
    [1,[2,[3,[4,[5,6,0]]]],8,9]
    """
    |> String.toLines
    |> String.concat "\n"

[<Theory>]
[<InlineData("[]", "[]")>]
[<InlineData("[1]", "[1]")>]
[<InlineData("[1,[12,3]]", "[1,[12,3]]")>]
[<InlineData("[[12,3],4]", "[[12,3],4]")>]
[<InlineData("[[12,3],[4,56]]", "[[12,3],[4,56]]")>]
[<InlineData("[[12,3],99,[4,56]]", "[[12,3],99,[4,56]]")>]
let ``Day13 parse works`` expected input =
    Assert.Equal(expected, input |> Data.parse |> Data.dump)

[<Fact>]
let ``Day13 part1 works`` () =
    Assert.Equal(13, input |> Data.parsePairs |> part1)

[<Fact>]
let ``Day13 part2 works`` () =
    Assert.Equal(140, input |> Data.parsePairs |> part2)

[<Fact>]
let ``Day13 Stars`` () =
    try
        Assert.Equal("5208", job1 ())
        Assert.Equal("25792", job2 ())
    with :? System.NotImplementedException ->
        ()
